generators/server/templates/src/main/java/package/config/_MicroserviceSecurityConfiguration.java

```diff
@@ -183,3 +183,127 @@ public RestTemplate vanillaRestTemplate() {
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+import org.springframework.security.config.http.SessionCreationPolicy;
+import org.springframework.security.oauth2.config.annotation.web.configuration.EnableResourceServer;
+import org.springframework.security.oauth2.config.annotation.web.configuration.ResourceServerConfigurerAdapter;
+import org.springframework.security.oauth2.provider.token.TokenStore;
+import org.springframework.security.oauth2.provider.token.store.JwtAccessTokenConverter;
+import org.springframework.security.oauth2.provider.token.store.JwtTokenStore;
+import org.springframework.security.web.util.matcher.RequestHeaderRequestMatcher;
+import org.springframework.security.web.util.matcher.RequestMatcher;
+import org.springframework.web.client.RestTemplate;
+
+import java.util.Map;
+import java.util.Optional;
+
+@Configuration
+@EnableResourceServer
+@EnableGlobalMethodSecurity(prePostEnabled = true, securedEnabled = true)
+public class MicroserviceSecurityConfiguration extends ResourceServerConfigurerAdapter {
+
+
+    private final ResourceServerProperties resourceServerProperties;
+
+    public MicroserviceSecurityConfiguration(ResourceServerProperties resourceServerProperties) {
+        this.resourceServerProperties = resourceServerProperties;
+    }
+
+    @Bean
+    @Primary
+    public UserInfoTokenServices userInfoTokenServices(PrincipalExtractor principalExtractor, AuthoritiesExtractor authoritiesExtractor) {
+        UserInfoTokenServices userInfoTokenServices = new UserInfoTokenServices(resourceServerProperties.getUserInfoUri(), resourceServerProperties.getClientId());
+        userInfoTokenServices.setPrincipalExtractor(principalExtractor);
+        userInfoTokenServices.setAuthoritiesExtractor(authoritiesExtractor);
+        return userInfoTokenServices;
+    }
+
+    @Bean
+    public PrincipalExtractor principalExtractor(@Value("${oauth2.principal-attribute}") String oauth2PrincipalAttribute) {
+        return new SimplePrincipalExtractor(oauth2PrincipalAttribute);
+    }
+
+    @Bean
+    public AuthoritiesExtractor authoritiesExtractor(@Value("${oauth2.authorities-attribute}") String oauth2AuthoritiesAttribute) {
+        return new SimpleAuthoritiesExtractor(oauth2AuthoritiesAttribute);
+    }
+
+<%_ if(applicationType === 'gateway') { _%>
+    @Bean
+    @Qualifier("authorizationHeaderRequestMatcher")
+    public RequestMatcher authorizationHeaderRequestMatcher() {
+        return new RequestHeaderRequestMatcher("Authorization");
+    }
+<%_ } _%>
+
+
+    @Override
+    public void configure(HttpSecurity http) throws Exception {
+        http
+            .csrf()
+            .disable()
+            .headers()
+            .frameOptions()
+            .disable()
+            .and()
+            .sessionManagement()
+            .sessionCreationPolicy(SessionCreationPolicy.STATELESS)
+            .and()
```

Comment:
I'd recommend out-denting the `.and`s in this code block, like https://github.com/jhipster/generator-jhipster/blob/master/generators/server/templates/src/main/java/package/config/_SecurityConfiguration.java#L191.